Recipes/InheritedTypeJsonConverter/InheritedTypeJsonConverter.cs

```diff
@@ -111,76 +111,31 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             }
 
             var jsonObject = JObject.Load(reader);
-            var childTypes = GetChildTypes(objectType);
-
             var jsonProperties = new HashSet<string>(GetProperties(jsonObject), StringComparer.OrdinalIgnoreCase);
-            var candidateChildTypes = new List<CandidateChildType>();
-            foreach (var childType in childTypes)
-            {
-                var childTypeProperties = childType.GetProperties().Select(t => t.Name).ToList();
-                var childTypeFields = childType.GetFields().Select(t => t.Name).ToList();
-                var childTypePropertiesAndFields = new HashSet<string>(
-                    childTypeProperties.Concat(childTypeFields),
-                    StringComparer.OrdinalIgnoreCase);
-                if (jsonProperties.All(p => childTypePropertiesAndFields.Contains(p)))
-                {
-                    var candidateChildType = new CandidateChildType
-                    {
-                        Type = childType,
-                        PropertiesAndFields = childTypePropertiesAndFields
-                    };
-                    candidateChildTypes.Add(candidateChildType);
-                }
-            }
 
-            foreach (var candidateChildType in candidateChildTypes)
-            {
-                try
-                {
-                    var deserializedObject = serializer.Deserialize(jsonObject.CreateReader(), candidateChildType.Type);
-                    if (deserializedObject != null)
-                    {
-                        candidateChildType.DeserializedObject = deserializedObject;
-                    }
-                }
-                catch (JsonException)
-                {
-                }
-            }
+            var candidateChildTypes = GetCandidateChildTypes(objectType, jsonProperties).ToList();
+
+            DeserializeCandidates(serializer, jsonObject, candidateChildTypes);
 
             candidateChildTypes = candidateChildTypes.Where(c => c.DeserializedObject != null).ToList();
```

Comment:
With my other suggestions this code can become: var candidateChildTypes = GetChildTypes().GetCandidates...(...).Deserialize...().Where(....).ToList();

Note that I would also name the methods more simply (and like a sentence) taking into context how they are intended to be called in sequence (unless I were trying to share them in a broader context).